drivers/include/mtd_mapper.h

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2020  Koen Zandberg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_mtd_mapper  MTD addres mapper
+ * @ingroup     drivers_storage
+ * @brief       Driver for address remap for flash devices
+ *
+ * This MTD module allows for remapping multiple different regions on a single
+ * MTD device and present them as separate MTD devices. This is similar to
+ * partitions on a hard drive, although this system only allows hardcoded
+ * partitions and lacks a partition table.
+ *
+ * The use case for this module is to be able to split a single MTD device, for
+ * example a SPI NOR flash chip into multiple separate regions which all can
+ * contain their own content or file systems.
+ *
+ * ## Usage
+ *
+ * To use this module include it in your makefile:
+ *
+ * ```
+ * USEMODULE += mtd_mapper
+ * ```
+ *
+ * To define new regions with an existing MTD device the following is required:
+ *
+ * ```
+ * mtd_mapper_parent_t parent = {
+ *     .mtd = MTD_0,
+ *     .mutex = MUTEX_INIT,
+ *     .init = false,
+ * };
```

Comment:
A macro might be handy to make sure `mutex` and `init` are properly initialized, like:

```
#define MTD_PARENT_INIT(_parent) \
{ \
    .mtd = _parent, \
    .mutex = MUTEX_INIT, \
    .init = false, \
}

Reply:
Good one, added the macro and used it in both the test application and the example code